### ISSUE TITLE
WPS114 Found underscored name pattern failes on cornercase

### ISSUE THREAD

**Issue by AlwxSin** [wemake-services/wemake-python-styleguide#1243]
# Bug report

## What's wrong
```python
# Correct:
star_wars_episode2 = 'awesome!'

# Incorrect
star_wars_2episode = "also awesome"
```
```
test.py
     1:   1 WPS114 Found underscored name pattern: star_wars_2episode [wemake-python-styleguide]
  star_wars_2episode = ""
  ^
```

## How is that should be
Checker should ignore numbers concatenated with strings.

There was a question [here](https://github.com/wemake-services/wemake-python-styleguide/issues/205#issuecomment-427339146) but with no answer

**Comment by sobolevn** [comment]
> Checker should ignore numbers concatenated with strings.

Why do you think so? 

The whole idea about this rule was to force `sw_episode2` instead of `sw_2episode`
You can also try `sw_2_episode`

**Comment by AlwxSin** [comment]
My personal problem is short abbreviations like `1c` or `360_security`.  I cant start a variable with a digit but I can't name it like `cotract_1c`  `firewall_360_security`

**Comment by sobolevn** [comment]
Let's make a PR with the experiment to allow this?

**Comment by AlwxSin** [comment]
@sobolevn I have made things. Allows characters after digit.
